auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -56,6 +58,8 @@
  * Wraps {@link Credentials} as a {@link CallCredentials}.
  */
 final class GoogleAuthLibraryCallCredentials implements CallCredentials {
+  private static final String SERVICE_ACCOUNT_CLASS_NAME
+      = "com.google.auth.oauth2.ServiceAccountCredentials";
```

Comment:
I am not a fan of this selective class loading.  It is going to make compiling hard internally without the dep, or force the dep to be present even if unused.

Please factor all uses of ServiceAccountCredentials to an individual Java file.